binaries/scheduler/main.go

```diff
@@ -6,38 +6,97 @@ import (
 	"log"
 
 	"github.com/apache/thrift/lib/go/thrift"
-	"github.com/scootdev/scoot/cloud/cluster/local"
-	"github.com/scootdev/scoot/sched/config"
+	"github.com/scootdev/scoot/config/json"
+	"github.com/scootdev/scoot/ice"
 	"github.com/scootdev/scoot/scootapi/server"
+
+	// For putting into ice.MagicBag
+	"github.com/scootdev/scoot/common/endpoints"
+	"github.com/scootdev/scoot/common/stats"
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched/distributor"
+	"github.com/scootdev/scoot/sched/scheduler"
+
+	// For putting into json.Options
+	"github.com/scootdev/scoot/config/scootconfig"
 )
 
 var addr = flag.String("addr", "localhost:9090", "Bind address for api server.")
 var httpPort = flag.Int("http_port", 9091, "port to serve http on")
 var cfgText = flag.String("sched_config", "", "Scheduler Configuration.")
 
+type servers struct {
+	thrift thrift.TServer
+	http   *endpoints.ObservabilityServer
+}
+
+func makeServers(thrift thrift.TServer, http *endpoints.ObservabilityServer) servers {
+	return servers{thrift, http}
+}
+
 func main() {
 	log.Println("Starting Cloud Scoot API Server & Scheduler")
 	flag.Parse()
 
-	parser := config.DefaultParser()
-	parser.Workers[""] = &config.RPCWorkersConfig{Type: "rpc"}
-	parser.Cluster[""] = &local.ClusterLocalConfig{Type: "local"}
-	parser.Cluster["local"] = &local.ClusterLocalConfig{}
-	parser.Report[""].(*config.DefaultReportConfig).HttpAddr = fmt.Sprintf("localhost:%d", *httpPort)
+	bag := ice.NewMagicBag()
+	bag.PutMany(
+		func() (thrift.TServerTransport, error) { return thrift.NewTServerSocket(*addr) },
+		endpoints.MakeStatsReceiver,
+		func(s stats.StatsReceiver) *endpoints.ObservabilityServer {
+			return endpoints.NewObservabilityServer(fmt.Sprintf(":%d", *httpPort), s)
+		},
+		makeServers,
+		saga.MakeSagaCoordinator,
+		saga.MakeInMemorySagaLog,
+		distributor.NewPoolDistributorFromCluster,
+		scheduler.NewScheduler,
+		server.StartScheduler,
+		server.MakeServer,
+		thrift.NewTTransportFactory,
+		func() thrift.TProtocolFactory {
+			return thrift.NewTBinaryProtocolFactoryDefault()
+		},
+	)
+
+	o := json.Options(map[string]json.Interface{
```

Comment:
You're right; this uses ice, but ice does not require it. (and it would be easy to change it work with another system)

I prefer jsonconfig and renamed it to that. Is that acceptable for now? Or can we think of something better?